plugins/device/rest.go

```diff
@@ -129,3 +129,33 @@ func deviceListHandler(w http.ResponseWriter, r *http.Request) {
 	restutils.SendHTTPResponse(ctx, w, http.StatusOK, devices)
 
 }
+
+func listAllDevicesHandler(w http.ResponseWriter, r *http.Request) {
+
+	ctx := r.Context()
+	logger := gdctx.GetReqLogger(ctx)
+	peerIDs, err := peer.GetPeerIDs()
+	if err != nil {
+		logger.WithError(err).Error("Failed to get peers from store")
+		restutils.SendHTTPError(ctx, w, http.StatusInternalServerError, err)
+		return
+	}
+
+	var devices []deviceapi.Info
+	for _, peerID := range peerIDs {
+		deviceList, err := deviceutils.GetDevices(peerID.String())
+		if err != nil {
+			logger.WithError(err).WithField("peerid", peerID).Error("Failed to get devices for peer")
+			if err == errors.ErrPeerNotFound {
```

Comment:
If it's `errors.ErrPeerNotFound`, do not treat is an error to be returned to client. As you have not taken locks (which is right), it's possible that a peer remove happens between line `137` and line `147`. Just log and `continue` if the peer is not found.